src/MenuLinkTreeOverride.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\stanford_profile_helper;
+
+use Drupal\Core\Menu\MenuLinkTreeInterface;
+use Drupal\Core\Menu\MenuTreeParameters;
+
+/**
+ * Service decorator for the menu.link_tree service.
+ */
+class MenuLinkTreeOverride implements MenuLinkTreeInterface {
```

Comment:
So, I think that if you do `extends MenuLinkTree` instead of implementing the whole interface, you can reduce this object to just the `build` function.  The rest of the object is essentially just calling the functions of the parent element, right?

Reply:
extending a service makes you rely on all the dependencies as well. If the service adds a new dependency argument, we'd have to modify this as well. It reduces the chance of the decorator service from breaking in the future.

This article explains that a little bit too https://www.blueoakinteractive.com/blog/service-decorators-drupal-8

Reply:
>>> Another option is to override and implement the entire class that is provided by the service. However, this means that your code is now responsible for providing the service to all other modules or services that depend on it.

Reply:
That's interesting.  I continue to be surprised by how drupal handles dependency injection; it's much different from other implementations.

Reply:
FWIW it's actually symfony: https://symfony.com/doc/current/service_container/service_decoration.html